js/src/components/chat/useChatState.ts

```diff
@@ -0,0 +1,177 @@
+import { useState, useCallback, useMemo } from "preact/hooks"
+import type { Message, UpdateUserInput } from "./types"
+
+export interface ChatState {
+  // State
+  messages: Message[]
+  inputValue: string
+  inputDisabled: boolean
+
+  // Message operations (for Shiny integration)
+  appendMessage: (message: Message) => void
+  appendMessageChunk: (message: Message) => void
+  clearMessages: () => void
+  removeLoadingMessage: () => void
+  updateUserInput: (update: UpdateUserInput) => void
```

Comment:
Seems like we could take some inspiration from #105 and re-work/simplify the event system?